torch/_functorch/_aot_autograd/fx_passes.py

```diff
@@ -0,0 +1,202 @@
+from collections import defaultdict
+from typing import Dict, List
+
+import torch
+from torch import fx
+from torch.utils import _pytree as pytree
+
+
+def is_primal(node: fx.Node) -> bool:
+    return node.op == "placeholder" and "tangents" not in str(node.target)
+
+
+def move_resize_zero_to_end_of_graph(graph: fx.Graph) -> None:
+    node_list = list(graph.nodes)
+    return_node = node_list[-1]
+    assert return_node.target == "output"
+    resize_nodes_to_insert_at_end = []
+    node_to_usage_map = defaultdict(list)
+    for i, n in enumerate(node_list):
+        if n.op == "call_function":
+            if (
+                n.target is torch.ops.inductor.resize_storage_bytes_.default
+                and n.args[1] == 0
```

Comment:
r.e. asserting more invariants: if we only plan on moving resizes to the end if they resize down to zero, then can we raise an assertion if we see an `inductor.resize_storage_bytes_()` for a non-zero size? (just so if we eventually learn we need to handle that case later, we get a fast not supported error)